modules/hostfrequencyscaling/hostfrequencyscaling.go

```diff
@@ -0,0 +1,516 @@
+		break
+	case "CPU_TEMP_CRITICALtoCPU_TEMP_NORMAL":
+		hfs.HostFrequencyScaling(me.NodeCfg, lowtoHighFreqScalPolicy.String(), bBoot)
+		break
+	case "CPU_TEMP_HIGHtoCPU_TEMP_NORMAL":
+		hfs.HostFrequencyScaling(me.NodeCfg, lowtoHighFreqScalPolicy.String(), bBoot)
+		break
+	}
+
+}
+
+// HostFrequencyScaling scales CPU frequency according to given parameters
+func (hfs *HFS) HostFrequencyScaling(node lib.Node, freqScalPolicy string, bBoot bool) {
+
+	freqScalPolicies := hfs.cfg.GetFreqScalPolicies()
+
+	scalingGovernor := freqScalPolicies[freqScalPolicy].GetScalingGovernor()
+	scalingMinFreq := freqScalPolicies[freqScalPolicy].GetScalingMinFreq()
+	scalingMaxFreq := freqScalPolicies[freqScalPolicy].GetScalingMaxFreq()
+
+	basePath := hfs.cfg.GetFreqSensorUrl() //"/sys/devices/system/cpu/cpufreq/policy0/"
+
+	// Set the CPU frequency scaling parameters
+	_ = ioutil.WriteFile(basePath+"scaling_governor", []byte(scalingGovernor), 0644)
+	_ = ioutil.WriteFile(basePath+"scaling_max_freq", []byte(scalingMaxFreq), 0644)
+	_ = ioutil.WriteFile(basePath+"scaling_min_freq", []byte(scalingMinFreq), 0644)
+
+	// Get the CPU frequency scaling parameters
+	bscalingGovernor, _ := ioutil.ReadFile(basePath + "scaling_governor")
+	bscalingMaxFreq, _ := ioutil.ReadFile(basePath + "scaling_max_freq")
+	bscalingMinFreq, _ := ioutil.ReadFile(basePath + "scaling_min_freq")
+
+	cpuCurFreq, _ := ioutil.ReadFile(basePath + "cpuinfo_cur_freq")
+	cpuMinFreq, _ := ioutil.ReadFile(basePath + "cpuinfo_min_freq")
+	cpuMaxFreq, _ := ioutil.ReadFile(basePath + "cpuinfo_max_freq")
+	scalingCurFreq, _ := ioutil.ReadFile(basePath + "scaling_cur_freq")
+
+	fscalingGovernor := strings.TrimSuffix(string(bscalingGovernor), "\n")
+
+	scalingMinFreqq := strings.TrimSuffix(string(bscalingMinFreq), "\n")
+	scalingMaxFreqq := strings.TrimSuffix(string(bscalingMaxFreq), "\n")
+	scalingCurFreqq := strings.TrimSuffix(string(scalingCurFreq), "\n")
+	cpuCurFreqq := strings.TrimSuffix(string(cpuCurFreq), "\n")
+	cpuMinFreqq := strings.TrimSuffix(string(cpuMinFreq), "\n")
+	cpuMaxFreqq := strings.TrimSuffix(string(cpuMaxFreq), "\n")
+
+	hostIP := GetNodeIPAddress()
+
+	currentScalingConfig := CPUPerfScalingResp{
+		TimeStamp:          time.Now(),
+		HostAddress:        hostIP,
+		CurScalingGovernor: fscalingGovernor,
+		ScalingMinFreq:     scalingMinFreqq,
+		ScalingMaxFreq:     scalingMaxFreqq,
+		ScalingCurFreq:     scalingCurFreqq,
+
+		CPUCurFreq: cpuCurFreqq,
+		CPUMinFreq: cpuMinFreqq,
+		CPUMaxFreq: cpuMaxFreqq,
+	}
+	fmt.Println("Host:", currentScalingConfig.HostAddress, "Scaling Governor:", currentScalingConfig.CurScalingGovernor, "Scaling Current Freq:", currentScalingConfig.ScalingCurFreq, "CPU Current Freq:", currentScalingConfig.CPUCurFreq, "Scaling Min Freq:", currentScalingConfig.ScalingMinFreq, "Scaling Max Freq:", currentScalingConfig.ScalingMaxFreq, "CPU Min Freq:", currentScalingConfig.CPUMinFreq, "CPU Max Freq:", currentScalingConfig.CPUMaxFreq)
+	fmt.Println("")
```

Comment:
Use hfs.api.Logf for this and use '\n' if you need an extra newline

Reply:
Fixed by commit 350d25b37ab43747487a4510feba5b17e9fce196